### PR TITLE
[Fix-8119][API] Update ProcessInstance error and without reason output

### LINKED ISSUE #8119: [Bug] [API] Update ProcessInstance error and without reason output
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

Update ProcessInstance error.And there is no cause for error.

### What you expected to happen

Update ProcessInstance error

### How to reproduce

create a new task in process instance edit page and link to another task.

### Version

2.0.2

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by SbloodyS**
## Purpose of the pull request

This pr will close #8119

### CODE REVIEW

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java [hunk 1]
```diff
@@ -1278,7 +1278,8 @@ private String recursionProcessDefinitionName(long projectCode, String processDe
             putMsg(result, Status.SUCCESS);
         } catch (Exception e) {
             result.put(Constants.STATUS, Status.REQUEST_PARAMS_NOT_VALID_ERROR);
-            result.put(Constants.MSG, e.getMessage());
+            putMsg(result, Status.REQUEST_PARAMS_NOT_VALID_ERROR, "");
+            logger.error(Status.REQUEST_PARAMS_NOT_VALID_ERROR.getMsg(), e);
```
Comment (ruanwenjun) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java]: I think use `Status.INTERNAL_SERVER_ERROR_ARGS` might be better.
```java
putMsg(result, Status.INTERNAL_SERVER_ERROR_ARGS, e.getMessage());
```
And it's suggested to find out why there is an exception here.

Comment (SbloodyS) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java]: I'll fix it.

**Review by caishunfeng — APPROVED** [review]
LGTM

**Review by ruanwenjun — APPROVED** [review]
LGTM

**Pull request merged**